wkcuber/converter.py

```diff
@@ -0,0 +1,283 @@
+            args.source_path = path.join(layer_path, mag)
+            args.layer_name = "color" if layer_path == "" else path.basename(layer_path)
+            args.name = dataset_name
+
+            convert_knossos(args)
+
+    def detect_dataset_and_layer_paths_with_mag(self) -> Tuple[str, Dict[str, str]]:
+        # Path structure for knossos is .../(dataset_name)/(layer_name)/(mag)folder/x0000/y0000/z0000/filename.raw
+        layers: Dict[str, set] = dict()
+        dataset_names = set()
+        for f in self.source_files:
+            split_path = path.normpath(f).split(sep)
+            starts_with_root = split_path[0] == ""
+            if starts_with_root:
+                split_path = split_path[1:]
+            assert (
+                len(split_path) >= 4
+            ), "Input Format is unreadable. Make sure to pass the path which points at least to a KNOSSOS magnification (e.g., testdata/knossos/color/1)."
+
+            if len(split_path) == 4:
+                # already inside the mag folder
+                add_or_update_if_present(layers, "", "")
+                dataset_names.add("dataset")
+            elif len(split_path) == 5:
+                # only the mag folder is given, therefore the layer path is empty
+                add_or_update_if_present(layers, "", split_path[0])
+                dataset_names.add("dataset")
+            elif len(split_path) == 6:
+                # additionally the layer folder is given, that should indicate a single layer as well
+                add_or_update_if_present(layers, split_path[0], split_path[1])
+                dataset_names.add("dataset")
+            else:
+                # also a dataset folder is given
+                layer_path = sep.join(split_path[0:-5])
+                if starts_with_root:
+                    layer_path = "/" + layer_path
+                add_or_update_if_present(layers, layer_path, split_path[-5])
+                dataset_names.add(split_path[-7])
+
+        assert (
+            len(dataset_names) == 1
+        ), "More than one dataset found. Stopping conversion..."
+        assert len(layers) > 0, "No layers found. Stopping conversion..."
+
+        layer_names_and_mags = dict()
+
+        for layer, mags in layers.items():
+            layer_names_and_mags[layer] = natsorted(list(mags))[0]
+
+        return dataset_names.pop(), layer_names_and_mags
+
+
+class ImageStackConverter(Converter):
+    def __init__(self) -> None:
+        self.source_files: List[str] = []
+        self.args: Namespace = Namespace()
+
+    def accepts_input(self, source_path: str) -> bool:
+        source_files = get_source_files(source_path, image_reader.readers.keys(), True)
+
+        if len(source_files) == 0:
+            return False
+
+        _, ext = path.splitext(source_files[0])
+
+        assert all(
+            map(lambda p: path.splitext(p)[1] == ext, source_files)
+        ), "Not all image files are of the same type"
```

Comment:
There was no similar assertion to that previously, but I was anxious to just allow this, since this order of processing depends on the file names, so there could be some inconsistencies. However, if you think this is not a problem, I'll gladly remove that check again.

Reply:
> There was no similar assertion to that previously,

Ok, then let's just keep it :)